vms/evm/imports_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package evm
+
+import (
+	"fmt"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"path/filepath"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestImportViolations ensures proper import rules:
+// - graft/coreth can be imported anywhere EXCEPT vms/evm (but vms/evm/emulate is an exception)
+// - graft/subnet-evm can only be imported within graft/subnet-evm itself and vms/evm/emulate
+// - graft/evm can NOT import from graft/coreth or graft/subnet-evm
+//
+// The rationale for these rules are as follows:
+//
+// coreth can be imported in AvalancheGo, because it was already imported by Avalanche prior to
+// grafting
+//
+// coreth can NOT be imported in the vms/evm package, because the goal is that vms/evm should
+// only contain the 'clean' properly uplifted code, that meets AvalancheGo quality standards.
+//
+// subnet-evm can NOT be imported anywhere in AvalancheGo besides graft/subnet-evm itself,
+// because it must not become a direct dependency of AvalancheGo or mix directly with coreth.
+//
+// graft/evm is the shared code that both coreth and subnet-evm will depend on, so it must not
+// import from either of them to maintain a one-way dependency direction.
+//
+// both coreth and subnet-evm can be imported in the vms/evm/emulate package, because it
+// allows consumers to use both coreth and subnet-evm registration at the same time.
+//
+// TODO(jonathanoppenheimer): remove the graft functionality once the graft package will be removed.
+func TestImportViolations(t *testing.T) {
+	const root = "../.."
+	repoRoot, err := filepath.Abs(root)
+	require.NoError(t, err)
+
+	graftDir := filepath.Join(repoRoot, "graft")
+	graftEVMDir := filepath.Join(repoRoot, "graft", "evm")
+	graftSubnetEVMDir := filepath.Join(repoRoot, "graft", "subnet-evm")
+	emulateDir := filepath.Join(repoRoot, "vms", "evm", "emulate")
+	vmsEVMDir := filepath.Join(repoRoot, "vms", "evm")
```

Comment:
```suggestion
	graftDir := filepath.Join(repoRoot, "graft")
	graftEVMDir := filepath.Join(graftDir, "evm")
	graftSubnetEVMDir := filepath.Join(graftDir, "subnet-evm")
	vmsEVMDir := filepath.Join(repoRoot, "vms", "evm")
	emulateDir := filepath.Join(vmsEVMDir "emulate")
```
Why not reuse definitions?